File: spark-study/src/main/scala/com/scala/RDDProgramming.scala
package com.scala

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext

object RDDProgramming extends App {
	val conf = new SparkConf().setAppName("appName").setMaster("local")
	val sc = new SparkContext(conf)

	val data = Array(1, 2)
	val distData = sc.parallelize(data)

	val logFile = "/usr/local/spark/README.md"
	val lines = sc.textFile(logFile)
	val lineLength = lines.map(s => s.length())
	val totalLength = lineLength.reduce((a, b) => a + b)
	println("totalLength:" + totalLength)

}

File: spark-study/src/main/java/com/RDDProgramming.java
package com;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;

public class RDDProgramming {

	public static void main(String[] args) {
		SparkConf conf = new SparkConf().setAppName("appname").setMaster("local");
		JavaSparkContext sc = new JavaSparkContext(conf);
		List<Integer> data = Arrays.asList(1, 2);
		JavaRDD<Integer> distData = sc.parallelize(data);

		String logFile = "README.md";
		JavaRDD<String> lines = sc.textFile(logFile);
		JavaRDD<Integer> lineLength = lines.map(s -> s.length());
		int totalLength = lineLength.reduce((a, b) -> a + b);
		System.out.println("totalLength:" + totalLength);
	}
}

class GetLength implements Function<String, Integer> {
	private static final long serialVersionUID = 1L;

	@Override
	public Integer call(String s) throws Exception {
		return s.length();
	}
}